extensions/line-lengths-3.3.md

```diff
@@ -0,0 +1,86 @@
+
+As an example, if `maxline` is 1024 then the maximum size of a full IRC message would be 2048 bytes (1024 for the tags, 1024 for the rest of the message).
+
+Servers MUST truncate incoming messages to their `maxline` values before processing them.
+
+`maxline` MUST be at least 512 and SHOULD default to at least 2048.
+
+### Interaction with PRIVMSG and NOTICE
+
+If a client has negotiated the `maxline` capability and sends a `PRIVMSG` or a `NOTICE` message that is longer than 512 octets, the receiving server MUST split this into multiple regular (512-octet) length messages when sending it to clients that have not negotiated the `maxline` capability.
+
+Servers SHOULD split on whitespace, but may use whatever method is easiest for them to implement. Splitting does not need to occur at the exact max length of the message, and servers can instead opt to split a number of characters earlier to simplify processing.
+
+Servers MAY split other commands/numerics into multiple lines in a way similar to `PRIVMSG` and `NOTICE` above, if it is purely for display purposes.
+
+### The `truncated` Tag
+
+The `truncated` tag, when present, indicates that a message has been truncated due to the client's line length. It may be sent to any client which supports message tags, as deemed appropriate by the IRCd.
+
+As an example, this tag may be used when a long channel `TOPIC` is set, but cannot correctly be relayed to the given user due to its length.
+
+### Examples
+
+In the following examples, the default message length is presumed to be 100 octets (rather than 512) and the extended length is presumed to be 400 octets (rather than 2048). This is done purely to simplify presentation and convey the concept more easily.
+
+* ` ->` conveys lines being sent from this client to the IRC server.
+* `<- ` conveys lines being sent from the server to this client.
+
+### Sending a long privmsg to someone else
+
+In this example, C1 and C2 have negotiated `maxlen`.
+
+    C1  -> PRIVMSG coolfriend :Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
+    C2 <-  :c1!test@localhost PRIVMSG coolfriend :Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
+
+In this example, C1 has negotiated `maxlen` but C2 has not.
+
+    C1  -> PRIVMSG coolfriend :Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
+    C2 <-  :c1!test@localhost PRIVMSG coolfriend :Lorem ipsum dolor sit amet, consectetur adipiscing elit,
```

Comment:
From the client perspective I feel like tingping, from the server perspective I don't mind providing a batch fallback when splitting lines.